File: SDKs/yodlee.com/1.1.0/python/src/sdk/models/shared/fieldoperation.py
import dataclasses
from typing import Any,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils

class FieldOperationFieldEnum(str, Enum):
    AMOUNT = "amount"
    DESCRIPTION = "description"

class FieldOperationOperationEnum(str, Enum):
    NUMBER_EQUALS = "numberEquals"
    NUMBER_LESS_THAN = "numberLessThan"
    NUMBER_LESS_THAN_EQUALS = "numberLessThanEquals"
    NUMBER_GREATER_THAN = "numberGreaterThan"
    NUMBER_GREATER_THAN_EQUALS = "numberGreaterThanEquals"
    STRING_EQUALS = "stringEquals"
    STRING_CONTAINS = "stringContains"


@dataclass_json
@dataclasses.dataclass
class FieldOperation:
    field: Optional[FieldOperationFieldEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('field') }})
    operation: Optional[FieldOperationOperationEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('operation') }})
    value: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('value') }})
    

File: SDKs/yodlee.com/1.1.0/typescript/src/sdk/models/shared/fieldoperation.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum FieldOperationFieldEnum {
    Amount = "amount",
    Description = "description"
}

export enum FieldOperationOperationEnum {
    NumberEquals = "numberEquals",
    NumberLessThan = "numberLessThan",
    NumberLessThanEquals = "numberLessThanEquals",
    NumberGreaterThan = "numberGreaterThan",
    NumberGreaterThanEquals = "numberGreaterThanEquals",
    StringEquals = "stringEquals",
    StringContains = "stringContains"
}


export class FieldOperation extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=field" })
  field?: FieldOperationFieldEnum;

  @SpeakeasyMetadata({ data: "json, name=operation" })
  operation?: FieldOperationOperationEnum;

  @SpeakeasyMetadata({ data: "json, name=value" })
  value?: Record<string, any>;
}
